src/frequenz/sdk/timeseries/formula_engine/_formula_steps.py

```diff
@@ -382,6 +428,9 @@ def stream_name(self) -> str:
         """
         return str(self._stream.__doc__)
 
+    def _value_is_invalid(self, value: QuantityT | None) -> bool:
+        return value is None or value.isnan() or value.isinf()
```

Comment:
I suggest using the positive form instead of negative, as a double negative could be hard to read (`if not self._value_is_invalid(...)` (and structure the method a bit more like a question), so:

```suggestion
    def _is_value_valid(self, value: QuantityT | None) -> bool:
        return value is not None and not value.isnan() and not value.isinf()
```

Reply:
done